src/database.test.js

```diff
@@ -1,161 +1,204 @@
-
-      expect(iterator.next().value)
-      .toEqual(call([ref, ref.push], data))
-
-      expect(context.app.database.mock.calls.length).toBe(1)
-      expect(context.app.database.mock.calls[0]).toEqual([])
-
-      expect(database.ref.mock.calls.length).toBe(1)
-      expect(database.ref.mock.calls[0]).toEqual([path])
-
-      expect(iterator.next(result)).toEqual({
-        done: true,
-        value: result.key
-      })
-    })
-  })
-
-  describe('update(path, data)', () => {
-    it('works', () => {
-      const path = 'skddksl'
-      const data = 'okqdkj'
-      const iterator = dbModule.update.call(context, path, data)
-
-      expect(iterator.next().value)
-      .toEqual(call([ref, ref.set], data))
-
-      expect(context.app.database.mock.calls.length).toBe(1)
-      expect(context.app.database.mock.calls[0]).toEqual([])
-
-      expect(database.ref.mock.calls.length).toBe(1)
-      expect(database.ref.mock.calls[0]).toEqual([path])
-
-      expect(iterator.next()).toEqual({
-        done: true,
-        value: undefined
-      })
-    })
-  })
-
-  describe('patch(path, data)', () => {
-    it('works', () => {
-      const path = 'skddksl'
-      const data = 'okqdkj'
-      const iterator = dbModule.patch.call(context, path, data)
-
-      expect(iterator.next().value)
-      .toEqual(call([ref, ref.update], data))
-
-      expect(context.app.database.mock.calls.length).toBe(1)
-      expect(context.app.database.mock.calls[0]).toEqual([])
-
-      expect(database.ref.mock.calls.length).toBe(1)
-      expect(database.ref.mock.calls[0]).toEqual([path])
-
-      expect(iterator.next()).toEqual({
-        done: true,
-        value: undefined
-      })
-    })
-  })
-
-  describe('delete(path)', () => {
-    it('works', () => {
-      const path = 'skddksl'
-      const iterator = dbModule.delete.call(context, path)
-
-      expect(iterator.next().value)
-      .toEqual(call([ref, ref.remove]))
-
-      expect(context.app.database.mock.calls.length).toBe(1)
-      expect(context.app.database.mock.calls[0]).toEqual([])
-
-      expect(database.ref.mock.calls.length).toBe(1)
-      expect(database.ref.mock.calls[0]).toEqual([path])
-
-      expect(iterator.next()).toEqual({
-        done: true,
-        value: undefined
-      })
-    })
-  })
-
-  describe('channel(path, event)', () => {
-    it('works', () => {
-      const path = 'skddksl'
-      const event = 'okqdkj'
-      dbModule.channel.call(context, path, event)
-
-      expect(context.app.database.mock.calls.length).toBe(1)
-      expect(context.app.database.mock.calls[0]).toEqual([])
-
-      expect(database.ref.mock.calls.length).toBe(1)
-      expect(database.ref.mock.calls[0]).toEqual([path])
-
-      expect(ref.on.mock.calls.length).toBe(1)
-      expect(ref.on.mock.calls[0][0]).toBe(event)
-    })
-  })
-})
+import { call } from 'redux-saga/effects'
+
+import dbModule from './database'
+
+describe('database', () => {

```

Comment:
Looks like the new commit was correctly diffed without all the line endings. I guess it might have been a git Windows anomaly?

Please review again.